### PR TITLE
Fix/16350 agent insertion wrong

### PR DESCRIPTION
**Pull request by GGP1**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/16350|

## Description

The parameters **ID** and **key** are now set separately in the arguments of the message sent to *authd*.

## Logs/Alerts example

Insert an agent using the 'Add agent full' endpoint:

```bash
-X POST https://localhost:55000/agents/insert \
    -H 'Content-Type: application/json' \
    -d '{"id": "001", "name": "test", "ip": "any"}'
```

The response id should match the one we specified above:

```json
{
    "data": {
        "id": "001",
        "key": "MDAxIHRlc3QgYW55IDU0YWNhOGQwNDYwMjg1MjI1Yzc1ZmUzY2U2ZjIwMGEwNjMyYzYyZTMwZDVkNGM1YjVkOTZjYjIwYzI3NzVkNjI="
    },
    "error": 0
}
```

## Tests

```
================================================================= test session starts =================================================================
platform linux -- Python 3.9.16, pytest-7.0.1, pluggy-1.0.0
rootdir: /home/gasti/work/wazuh/framework
plugins: anyio-3.6.2, asyncio-0.18.1
asyncio: mode=legacy
collected 1649 items                                                                                                                                  

framework/wazuh/core/cluster/dapi/tests/test_dapi.py ...........................                                                                [  1%]
framework/wazuh/core/cluster/tests/test_cluster.py ...................                                                                          [  2%]
framework/wazuh/core/cluster/tests/test_common.py .......                                                                                       [  3%]
framework/wazuh/core/cluster/tests/test_control.py .....                                                                                        [  3%]
framework/wazuh/core/cluster/tests/test_local_client.py .                                                                                       [  3%]
framework/wazuh/core/cluster/tests/test_utils.py .......                                                                                        [  4%]
framework/wazuh/core/cluster/tests/test_worker.py .......                                                                                       [  4%]
framework/wazuh/core/tests/test_active_response.py ..............                                                                               [  5%]
framework/wazuh/core/tests/test_agent.py ...................................................................................................... [ 11%]
.........................................                                                                                                       [ 13%]
framework/wazuh/core/tests/test_cdb_list.py ......................................                                                              [ 16%]
framework/wazuh/core/tests/test_common.py .......                                                                                               [ 16%]
framework/wazuh/core/tests/test_configuration.py ....................................                                                           [ 18%]
framework/wazuh/core/tests/test_database.py .............                                                                                       [ 19%]
framework/wazuh/core/tests/test_decoder.py ................                                                                                     [ 20%]
framework/wazuh/core/tests/test_exception.py ...                                                                                                [ 20%]
framework/wazuh/core/tests/test_input_validator.py ...                                                                                          [ 20%]
framework/wazuh/core/tests/test_logtest.py ..                                                                                                   [ 21%]
framework/wazuh/core/tests/test_manager.py ...............                                                                                      [ 22%]
framework/wazuh/core/tests/test_mitre.py .............                                                                                          [ 22%]
framework/wazuh/core/tests/test_pyDaemonModule.py .....                                                                                         [ 23%]
framework/wazuh/core/tests/test_results.py ........................................                                                             [ 25%]
framework/wazuh/core/tests/test_rootcheck.py .............                                                                                      [ 26%]
framework/wazuh/core/tests/test_rule.py ......................                                                                                  [ 27%]
framework/wazuh/core/tests/test_sca.py ............                                                                                             [ 28%]
framework/wazuh/core/tests/test_security.py ............                                                                                        [ 29%]
framework/wazuh/core/tests/test_stats.py ............                                                                                           [ 29%]
framework/wazuh/core/tests/test_syscheck.py .......                                                                                             [ 30%]
framework/wazuh/core/tests/test_syscollector.py ...                                                                                             [ 30%]
framework/wazuh/core/tests/test_task.py ........                                                                                                [ 30%]
framework/wazuh/core/tests/test_utils.py ...................................................................................................... [ 37%]
..............................................................................................................................                  [ 44%]
framework/wazuh/core/tests/test_vulnerability.py ..                                                                                             [ 44%]
framework/wazuh/core/tests/test_wazuh_queue.py ..................                                                                               [ 45%]
framework/wazuh/core/tests/test_wazuh_socket.py ....................                                                                            [ 47%]
framework/wazuh/core/tests/test_wdb.py ......................                                                                                   [ 48%]
framework/wazuh/core/tests/test_wlogging.py ........                                                                                            [ 48%]
framework/wazuh/rbac/tests/test_auth_context.py ..                                                                                              [ 49%]
framework/wazuh/rbac/tests/test_decorators.py ................................................................................................. [ 55%]
........                                                                                                                                        [ 55%]
framework/wazuh/rbac/tests/test_default_configuration.py .......................................................                                [ 58%]
framework/wazuh/rbac/tests/test_orm.py ......................................................                                                   [ 62%]
framework/wazuh/rbac/tests/test_preprocessor.py ...........                                                                                     [ 62%]
framework/wazuh/tests/test_active_response.py ............                                                                                      [ 63%]
framework/wazuh/tests/test_agent.py ........................................................................................................... [ 69%]
..........                                                                                                                                      [ 70%]
framework/wazuh/tests/test_cdb_list.py .....................................................                                                    [ 73%]
framework/wazuh/tests/test_ciscat.py .................................                                                                          [ 75%]
framework/wazuh/tests/test_cluster.py ..........                                                                                                [ 76%]
framework/wazuh/tests/test_decoder.py ...................................                                                                       [ 78%]
framework/wazuh/tests/test_group.py ........                                                                                                    [ 79%]
framework/wazuh/tests/test_logtest.py ......                                                                                                    [ 79%]
framework/wazuh/tests/test_manager.py ....................................                                                                      [ 81%]
framework/wazuh/tests/test_mitre.py .......                                                                                                     [ 81%]
framework/wazuh/tests/test_rootcheck.py ..................................................                                                      [ 85%]
framework/wazuh/tests/test_rule.py ........................................................                                                     [ 88%]
framework/wazuh/tests/test_sca.py .......                                                                                                       [ 88%]
framework/wazuh/tests/test_security.py .........................................................................                                [ 93%]
framework/wazuh/tests/test_stats.py .......                                                                                                     [ 93%]
framework/wazuh/tests/test_syscheck.py ..........................                                                                               [ 95%]
framework/wazuh/tests/test_syscollector.py ............                                                                                         [ 95%]
framework/wazuh/tests/test_task.py ............................                                                                                 [ 97%]
framework/wazuh/tests/test_vulnerability.py ......................................                                                              [100%]

==================================================== 1649 passed, 34 warnings in 228.98s (0:03:48) ====================================================
```

### CODE REVIEW

File: framework/wazuh/core/tests/test_agent.py [hunk 1]
```diff
@@ -696,6 +696,7 @@ def test_agent_add_ko(mock_maganer_status):
 
 @pytest.mark.parametrize("name, ip, id, key, force", [
     ('test_agent', '172.19.0.100', None, None, None),
+    ('test_agent', 'any', '001', None, None),
```
Comment (nico-stefani) [inline framework/wazuh/core/tests/test_agent.py:699]: I think we need a test case for the combination `id=None` and `key=<some-value>`

**Review by nico-stefani — CHANGES_REQUESTED** [review]

File: framework/wazuh/core/tests/test_agent.py (hunk 1 above)
Comment (GGP1) [reply framework/wazuh/core/tests/test_agent.py:699]: Agreed, I will add that as well

**Review by nico-stefani — DISMISSED** [review]

**Review by RamosFe — DISMISSED** [review]

**Review by nico-stefani — APPROVED** [review]
